src/pages/dataTableUtilities/reducer/tableReducers.js

```diff
@@ -175,13 +175,19 @@ export const toggleStockOut = state => {
  * Also removes the current sorting and filter.
  */
 export const addRecord = (state, action) => {
-  const { backingData } = state;
+  const { data, searchTerm, backingData } = state;
   const { payload } = action;
   const { record } = payload;
 
+  // New records are added to index 0 of the table and all filtering and
+  // sorting are removed. To keep the table consistent while adding items,
+  // if there is a search term, the table is currently being filtered so
+  // fetch all the rows again. Otherwise, just use the current data state.
+  const newData = searchTerm ? backingData.slice(1, backingData.length - 1) : data;
```

Comment:
From me trying to understand, you're slicing from index 1 because `backingData[0]` should already be the new record and otherwise you'd show it twice when adding it on line 190 (note the old implementation didn't do this)? If that is true... couldn't just use `backingData` directly without any logic? If it's not true, then a line is getting lost?

The second parameter to [slice](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/slice) is optional and should automatically slice the rest of the array without giving an argument. 
```suggestion
  const newData = searchTerm ? backingData.slice(1) : data;
```
Depending on first comments, the first parameter is also optional too... But then not sure why you need slicing at all when you're spreading?

Reply:
hm you're right that was stupid. I did this instead. Maybe also stupid 611df7586450b6271c90060a5deab019dc396d7c -